### PR TITLE
Fix: S3 read text with version ID was not working

### PR DESCRIPTION
**Pull request by LeonLuttenberger**
### Feature or Bugfix
- Bugfix

### Detail
In #1567, the ability to fetch a specific version of an S3 object was broken. The unit tests in that were meant to cover this use-case were not checking the code well enough, as they were only checking whether the most recent version had been pulled.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubLoadTests5656BB24-s6u9F3qN9oFy
* Commit ID: 23af64f1f0cfdc234c61c7a89df69058b628584e
* Result: FAILED
* [Build Logs](https://pciyom2pbh.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=9817921d-8498-4fac-8319-11909d8607e5%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubLoadTests5656BB24-s6u9F3qN9oFy
* Commit ID: ea999efe4804cd05f12effaf79e1f0c203baf1d8
* Result: FAILED
* [Build Logs](https://pciyom2pbh.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=8f378de0-e7ef-4683-8006-c0a20f77e4fe%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: 23af64f1f0cfdc234c61c7a89df69058b628584e
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=327660bc-5c5c-4585-8132-682a070c66c0%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubStandardCodeBuild8C06-llutOAimTATs
* Commit ID: 23af64f1f0cfdc234c61c7a89df69058b628584e
* Result: FAILED
* [Build Logs](https://jsv9u8gote.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=3aceb10e-8cdd-42b0-a722-837c8fa38aa6%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubLoadTests5656BB24-s6u9F3qN9oFy
* Commit ID: ea999efe4804cd05f12effaf79e1f0c203baf1d8
* Result: FAILED
* [Build Logs](https://pciyom2pbh.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=598de766-5b2c-4ba8-a791-2590d73c0b51%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by kukushking — APPROVED** [review]

File: awswrangler/s3/_read_text.py [hunk 1]
```diff
@@ -33,6 +33,16 @@
 _logger: logging.Logger = logging.getLogger(__name__)
 
 
+def _get_version_id_for(version_id: Optional[Union[str, Dict[str, str]]], path: str) -> Optional[str]:
+    if version_id is None:
+        return None
```
[the remark this answers is not in the document]
Comment (LeonLuttenberger) [reply awswrangler/s3/_read_text.py:38]: True, good catch.

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubLoadTests5656BB24-s6u9F3qN9oFy
* Commit ID: 31557e4ac1835d1887843da208379d9d2dc86cfe
* Result: FAILED
* [Build Logs](https://pciyom2pbh.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=1a4eb0e4-f667-4206-bdeb-480007a1dc07%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: 31557e4ac1835d1887843da208379d9d2dc86cfe
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=3930ddc7-22ac-4ea9-93e8-a4dd2b5d659a%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubStandardCodeBuild8C06-llutOAimTATs
* Commit ID: 31557e4ac1835d1887843da208379d9d2dc86cfe
* Result: FAILED
* [Build Logs](https://jsv9u8gote.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=283dd27e-6763-4238-b3e0-e7e968ff4976%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by malachi-constant — APPROVED** [review]

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: 9e3dd0dc6c29e31c3ce1088493f3fadd3d77c1d2
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=d03c4189-dc76-4833-8a7c-80d0bf803511%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubLoadTests5656BB24-s6u9F3qN9oFy
* Commit ID: 9e3dd0dc6c29e31c3ce1088493f3fadd3d77c1d2
* Result: SUCCEEDED
* [Build Logs](https://pciyom2pbh.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=cb91664c-40bd-4dec-8be4-d767c4845be5%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubStandardCodeBuild8C06-llutOAimTATs
* Commit ID: 9e3dd0dc6c29e31c3ce1088493f3fadd3d77c1d2
* Result: FAILED
* [Build Logs](https://jsv9u8gote.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=d651f4b4-fcac-4476-b40f-d49c644a52f4%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubStandardCodeBuild8C06-llutOAimTATs
* Commit ID: 9e3dd0dc6c29e31c3ce1088493f3fadd3d77c1d2
* Result: SUCCEEDED
* [Build Logs](https://jsv9u8gote.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=6f790319-e9f3-4264-abda-964cc289ee7c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**